src/Monolog/Logger.php

```diff
@@ -329,7 +338,7 @@ public function useLoggingLoopDetection(bool $detectCycles): self
      *
      * @phpstan-param value-of<Level::VALUES>|Level $level
      */
-    public function addRecord(int|Level $level, string $message, array $context = [], JsonSerializableDateTimeImmutable|null $datetime = null): bool
+    public function addRecord(int|Level $level, string $message, array $context = [], bool $useClock = false): bool
```

Comment:
This signature change is a BC break, which is a no-go in a minor version (and a bad idea in a major version if we don't provide a migration path through deprecations first)